src/main/java/net/glowstone/block/blocktype/BlockCarpet.java

```diff
@@ -6,6 +6,6 @@
 public class BlockCarpet extends BlockNeedsAttached {
     @Override
     public boolean canPlaceAt(GlowBlock block, BlockFace against) {
-        return super.canPlaceAt(block, against) && !block.getRelative(BlockFace.DOWN).isEmpty();
+        return !block.getRelative(BlockFace.DOWN).isEmpty();
```

Comment:
Yes, it should be changed everywhere, especially in BlockLiquid [eg](https://github.com/GlowstoneMC/GlowstonePlusPlus/blob/master/src/main/java/net/glowstone/block/blocktype/BlockLiquid.java#L133)